mypy/semanal.py

```diff
@@ -2120,14 +2120,17 @@ def analyze_namedtuple_assign(self, s: AssignmentStmt) -> bool:
         """Check if s defines a namedtuple."""
         if isinstance(s.rvalue, CallExpr) and isinstance(s.rvalue.analyzed, NamedTupleExpr):
             return True  # This is a valid and analyzed named tuple definition, nothing to do here.
-        if len(s.lvalues) != 1 or not isinstance(s.lvalues[0], NameExpr):
+        if len(s.lvalues) != 1 or type(s.lvalues[0]) not in [NameExpr, MemberExpr]:
             return False
         lvalue = s.lvalues[0]
         name = lvalue.name
         is_named_tuple, info = self.named_tuple_analyzer.check_namedtuple(s.rvalue, name,
                                                                           self.is_func_scope())
         if not is_named_tuple:
             return False
+        if isinstance(s.lvalues[0], MemberExpr):
+            self.fail("NamedTuple type as an attribute is not supported", s.lvalues[0])
+            return False
```

Comment:
This hole block could be moved at the start of function before `check_namedtuple` is called.   
but then strangely a `error: "int" not callable` error happens during test-case.   
i believe the error message is totally miss-leading.

Reply:
The `"int" not callable` error is due to how test stubs cheat a bit and use too simple definitions for some things such as `NamedTuple`. I'd recommend moving the new test case to `test-data/unit/pythoneval.test` which uses the full typeshed stubs. This should allow you to move this to the start of the function.

Reply:
I moved the block to the start, but `NamedTuple type as a ...` error was generated on every line with an attribute assignment statment.  
i think this error should be created only if RHS is a NamedTuple so the if is better at the end.